### PR TITLE
Simplified res model

### PR DESCRIPTION
**Pull request by sondrev**

### CODE REVIEW

File: search-commons/src/main/java/no/unit/nva/search/common/ParameterValidator.java [hunk 1]
```diff
@@ -185,7 +185,7 @@ protected String getDecodedValue(ParameterKey<K> qpKey, String value) {
 
     /** Adds query and path parameters from requestInfo. */
     public ParameterValidator<K, Q> fromRequestInfo(RequestInfo requestInfo) {
-        searchQuery.setMediaType(requestInfo.getHeaders().get(ACCEPT));
+        searchQuery.setMediaType(ContentTypeUtils.extractMimeTypeFromRequestInfo(requestInfo));
```
Comment (StigNorland) [inline search-commons/src/main/java/no/unit/nva/search/common/ParameterValidator.java:188]: Kan du importere `extractMimeTypeFromRequestInfo` direkte?

File: search-commons/src/main/java/no/unit/nva/search/resource/SimplifiedResourceModelMutator.java [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+        return new Series(series.path(ID).textValue(), series.path(NAME).textValue());
+    }
+
+    private List<Contributor> mutateContributorsPreview(JsonNode source) {
+        var contributors = new ArrayList();
+        source.path(ENTITY_DESCRIPTION)
+                .path(CONTRIBUTORS_PREVIEW)
+                .iterator()
+                .forEachRemaining(
+                        c -> {
```
Comment (StigNorland) [inline search-commons/src/main/java/no/unit/nva/search/resource/SimplifiedResourceModelMutator.java:278]: rename to contributor or something with a better description.

File: search-commons/src/main/java/no/unit/nva/search/resource/SimplifiedResourceModelMutator.java [hunk 3]
```diff
@@ -0,0 +1,300 @@
...
+    public static final String CONTRIBUTORS_PREVIEW = "contributorsPreview";
+    public static final String CORRESPONDING_AUTHOR = "correspondingAuthor";
+    public static final String IDENTITY = "identity";
+    public static final String SEQUENCE = "sequence";
+    public static final String ROLE = "role";
+    private final ObjectMapper objectMapper = dtoObjectMapper.copy();
+    public static final String ENTITY_DESCRIPTION = "entityDescription";
+    public static final String REFERENCE = "reference";
+    public static final String PUBLICATION_CONTEXT = "publicationContext";
+    public static final String SERIES = "series";
```
Comment (StigNorland) [inline search-commons/src/main/java/no/unit/nva/search/resource/SimplifiedResourceModelMutator.java:76]: many of these are already defined in Constants.Words....

File: search-commons/src/main/java/no/unit/nva/search/resource/SimplifiedResourceModelMutator.java
Comment (StigNorland) [inline search-commons/src/main/java/no/unit/nva/search/resource/SimplifiedResourceModelMutator.java:1]: The overall structure, i.e readability is high 👍

File: search-commons/src/main/java/no/unit/nva/search/resource/response/Affiliation.java [hunk 4]
```diff
@@ -1,12 +1,3 @@
...
-
-class Affiliation {
-
-    @JsonProperty("id")
-    private String id;
-
-    @JsonProperty("type")
-    private String type;
-}
+public record Affiliation(String id, String type) {}
```
Comment (StigNorland) [inline search-commons/src/main/java/no/unit/nva/search/resource/response/Affiliation.java:3]: thank you!

File: search-commons/src/main/java/no/unit/nva/search/resource/response/Contributor.java [hunk 5]
```diff
@@ -1,25 +1,9 @@
...
-    @JsonProperty("role")
-    private String role;
-
-    @JsonProperty("sequence")
-    private int sequence;
-
-    @JsonProperty("type")
-    private String type;
-}
+public record Contributor(
```
Comment (StigNorland) [inline search-commons/src/main/java/no/unit/nva/search/resource/response/Contributor.java:3]: 👍

File: search-handlers/src/test/java/no/unit/nva/search/SearchResource20241201HandlerTest.java [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+        when(mockedSearchClient.doSearch(any())).thenReturn(body);
+    }
+
+    @Test
+    public void shouldReturnAResponseThatCanBeMappedToModelDto() throws IOException {
+        prepareRestHighLevelClientOkResponse();
+        handler.handleRequest(getInputStream(), outputStream, contextMock);
+        //        var gatewayResponse =
+        //                GatewayResponse.fromOutputStream(outputStream,
+        // ResourceSearchResponse.class);
```
Comment (StigNorland) [inline search-handlers/src/test/java/no/unit/nva/search/SearchResource20241201HandlerTest.java:73]: remove?

**Review by StigNorland — CHANGES_REQUESTED** [review]

File: search-commons/src/main/java/no/unit/nva/search/resource/SimplifiedResourceModelMutator.java [hunk 7]
```diff
@@ -0,0 +1,300 @@
...
+                                source.path(ENTITY_DESCRIPTION)
+                                        .path(REFERENCE)
+                                        .path(PUBLICATION_CONTEXT)
+                                        .path(SERIES)
+                                        .path(PRINT_ISSN)
+                                        .textValue())
+                        .filter(Objects::nonNull)
+                        .collect(Collectors.toSet());
+
+        var isbnsInSource =
```
Comment (brinxmat) [inline search-commons/src/main/java/no/unit/nva/search/resource/SimplifiedResourceModelMutator.java:185]: Unfortunately, we also have ISBN in Manifestations of Artistic

File: search-commons/src/main/java/no/unit/nva/search/resource/SimplifiedResourceModelMutator.java [hunk 8]
```diff
@@ -0,0 +1,300 @@
...
+                                    break;
+                            }
+                        });
+
+        return new OtherIdentifiers(
+                new HashSet<>(scopusIdentifiers),
+                new HashSet<>(cristinIdentifiers),
+                new HashSet<>(handleIdentifiers),
+                new HashSet<>(isbns),
+                new HashSet<>(issns));
```
Comment (brinxmat) [inline search-commons/src/main/java/no/unit/nva/search/resource/SimplifiedResourceModelMutator.java:222]: These are already Sets

File: search-commons/src/main/java/no/unit/nva/search/resource/SimplifiedResourceModelMutator.java [hunk 9]
```diff
@@ -0,0 +1,300 @@
...
+                        .path(PUBLICATION_CONTEXT)
+                        .path(ISBN_LIST);
+        List<String> isbns =
+                isbnsInSource.isMissingNode()
+                        ? Collections.emptyList()
+                        : objectMapper.readerForListOf(String.class).readValue(isbnsInSource);
+
+        var handleIdentifiers = new ArrayList<String>();
+        var cristinIdentifiers = new ArrayList<String>();
+        var scopusIdentifiers = new ArrayList<String>();
```
Comment (brinxmat) [inline search-commons/src/main/java/no/unit/nva/search/resource/SimplifiedResourceModelMutator.java:197]: Is there a reason to not use HashSet here?

File: search-commons/src/main/java/no/unit/nva/search/resource/SimplifiedResourceModelMutator.java [hunk 10]
```diff
@@ -0,0 +1,300 @@
...
+                path(ENTITY_DESCRIPTION, PUBLICATION_DATE));
+    }
+
+    @Override
+    public JsonNode transform(JsonNode source) {
+        return (JsonNode)
+                attempt(() -> objectMapper.valueToTree(transformToDto(source))).orElseThrow();
+    }
+
+    private ResourceSearchResponse transformToDto(JsonNode source) throws IOException {
```
Comment (brinxmat) [inline search-commons/src/main/java/no/unit/nva/search/resource/SimplifiedResourceModelMutator.java:111]: I'd extract every possible method with a good name as this is reasonably complex.

File: search-commons/src/main/java/no/unit/nva/search/resource/SimplifiedResourceModelMutator.java [hunk 11]
```diff
@@ -0,0 +1,300 @@
...
+                source.path(ENTITY_DESCRIPTION)
+                        .path(REFERENCE)
+                        .path(PUBLICATION_CONTEXT)
+                        .path(NAME)
+                        .textValue(),
+                source.path(ENTITY_DESCRIPTION).path(REFERENCE).path(DOI).textValue());
+    }
+
+    private String mutateJournalOrPublisher(JsonNode source) {
+        return JOURNAL.equals(
```
Comment (brinxmat) [inline search-commons/src/main/java/no/unit/nva/search/resource/SimplifiedResourceModelMutator.java:249]: I think we have other possibilities

File: search-handlers/src/main/java/no/unit/nva/search/SearchResource20241201Handler.java [hunk 12]
```diff
@@ -0,0 +1,79 @@
...
+                .requiredStatus(PUBLISHED, PUBLISHED_METADATA)
+                .apply()
+                .doSearch(opensearchClient)
+                .withMutators(new SimplifiedResourceModelMutator())
+                .toString();
+    }
+
+    @Override
+    protected Integer getSuccessStatusCode(Void input, String output) {
+        return HttpURLConnection.HTTP_OK;
```
[the remark this answers is not in the document]
Comment (sondrev) [reply search-handlers/src/main/java/no/unit/nva/search/SearchResource20241201Handler.java:77]: @brinxmat Can you explain furhter? I dont follow

File: search-commons/src/main/java/no/unit/nva/search/resource/SimplifiedResourceModelMutator.java (hunk 7 above)
Comment (sondrev) [reply search-commons/src/main/java/no/unit/nva/search/resource/SimplifiedResourceModelMutator.java:185]: @brinxmat Can you help me with the jsonpath for those? cant find them

File: search-commons/src/main/java/no/unit/nva/search/resource/SimplifiedResourceModelMutator.java [hunk 13]
```diff
@@ -0,0 +1,300 @@
...
+                .iterator()
+                .forEachRemaining(
+                        c -> {
+                            var affiliationNode = c.path(AFFILIATIONS);
+                            var affiliation =
+                                    affiliationNode.isMissingNode()
+                                            ? null
+                                            : new Affiliation(
+                                                    affiliationNode.get(0).path(ID).textValue(),
+                                                    affiliationNode.get(0).path(TYPE).textValue());
```
Comment (sondrev) [inline search-commons/src/main/java/no/unit/nva/search/resource/SimplifiedResourceModelMutator.java:285]: @brinxmat Is this correct? In the full model a Contributor has multiple affiliations. Do we only want 1 or should I keep all? Do we need both id and type?

**Pull request merged**